lib/propshaft/asset.rb

```diff
@@ -16,12 +19,27 @@ def content_type
     Mime::Type.lookup_by_extension(logical_path.extname.from(1))
   end
 
+  def may_depend?
+    content_type&.symbol == :css
+  end
+
   def length
     content.size
   end
 
+  # A dependency aware digest can be calculated for any asset that has no dependencies or
+  # that has had its dependency fingerprint set by DependencyTree.
+  # If it's too soon, we can still calculate the digest based on file contents,
+  # but there won't be any dependency cache busting.
+  def digest_too_soon?
+    may_depend? && dependencies&.any? && !dependency_fingerprint
+  end
+
   def digest
-    @digest ||= Digest::SHA1.hexdigest("#{content}#{version}").first(8)
+    @digest ||= begin
+      Propshaft.logger.warn("digest dependencies not ready for #{logical_path}") if digest_too_soon?
```

Comment:
Maybe I'm missing something obvious, but in what scenario would an asset have dependencies but no dependency fingerprint? It's just not clear to me what should be done by users in response to this warning, so either failing more loudly (and clearly) or not warning at all seems like better options.

Reply:
That should only happen if there was a bug in the dependency tree logic. Since digest bugs are insidious, I decided a warning was necessary, but going to the extent of failing a build for inadequate cache busting seemed a little much.